packages/costa/src/runtime.ts

```diff
@@ -81,22 +81,20 @@ function extractPackageJsonFromReadable(readable: NodeJS.ReadableStream, pkgFile
   return new Promise((resolve, reject) => {
     let packageJson = '';
     const extract = tar.extract();
-    extract.on('error', reject);
     extract.on('entry', (header, stream, next) => {
       if (header.name === pkgFilename) {
         stream.on('data', (buf) => packageJson += buf);
       }
       stream.once('end', next);
       stream.resume();
     });
-    extract.on('finish', () => {
+    pipeWithFinish(readable, extract).then(() => {
```

Comment:
No necessary to return a Promise here, we could use a callback style and call promisify for async/await use-case.

Reply:
This part of the code could also be simplified a little bit.

Reply:
Updated.